### ISSUE TITLE
Autopilot not working

### ISSUE THREAD

**Issue by jorgemarey** [hashicorp/nomad#4210]
### Nomad version
```
Nomad v0.8.1 (46aa11ba851f114f5faa0876839b45fbca0c47f0)
```

### Operating system and Environment details
```
4.9.0-6-amd64 #1 SMP Debian 4.9.82-1+deb9u3 (2018-03-02) x86_64 GNU/Linux
```

### Issue
When I launch a server in dev mode (or normal mode) the execution of the following command fails
```
$ nomad operator autopilot set-config -cleanup-dead-servers=true
Configuration could not be atomically updated, please try again
```

If I run the get-config command, all values are not the defaults indicated on the [documentation](https://www.nomadproject.io/docs/agent/configuration/autopilot.html)
```
$ nomad operator autopilot get-config
CleanupDeadServers = false
LastContactThreshold = 0s
MaxTrailingLogs = 0
ServerStabilizationTime = 0s
EnableRedundancyZones = false
DisableUpgradeMigration = false
EnableCustomUpgrades = false
```

It seems that they all default to the golang default type value.

I cloned and made checkout of the v0.8.1 tag and several tests failed. All related to autopilot. 

I found that the structs on the file https://github.com/hashicorp/nomad/blob/master/nomad/structs/operator.go
where not generated with **codecgen**. 

I made the following change on https://github.com/hashicorp/nomad/blob/master/nomad/structs/structs_codegen.go

//go:generate codecgen -d 100 -o structs.generated.go structs.go *operator.go*

Then I compiled and executed and the autopilot commands seems to have worked fine, but I don't know If there's something else needed for this to work properly.

Thanks.

**Comment by chelseakomlo** [comment]
Thanks for the detailed reproduction steps, I've confirmed this issue. We'll have a fix for this in our next 0.8.2 release.

**Comment by jorgemarey** [comment]
Awesome! Thanks 😃
